### DISCUSSION TITLE
error deploying aws s3 to vercel

### DISCUSSION THREAD

**Discussion by quyenvnk63** [vercel/next.js#51847] [Help]
### Summary

Hi everybody. I have a file like the following[ const AWS = require('aws-sdk');
const { v4: uuidv4 } = require('uuid');
require('dotenv').config();
const bucket = 'up-load-url';
const s3 = new AWS.S3({
   accessKeyId: process.env.AWS_ACCESS_KEY_ID,
   secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
   region: process.env.AWS_DEFAULT_REGION,
   s3ForcePathStyle: process.env.AWS_USE_PATH_STYLE_ENDPOINT === 'false',
   signatureVersion: 'v4'
});

async function getPresignedUrl(contentType) {
   const ex = contentType.split("/")[1];
   const key = `${uuidv4()}.${ex}`;
   const params = {
     Bucket: bucket,
     Key: key,
     ContentType: contentType,
     Expires: 600,
   };
   const signedUrl = await s3.getSignedUrlPromise('putObject', params);
   return { key, url: signedUrl };
}
]. I get the error message as [
Vercel — Deployment failed.]. when i call the api, the error message is [ "error": "Missing required key 'Bucket' in params"]. i have successfully run it locally.i tried setting variable in vercel but still no success. i need your help. thank you.

**Comment by icyJoseph** [comment]
Does the S3 bucket with name, 'up-load-url' exist on that S3 bucket?

Double check that your `AWS_` env vars are properly set in the Vercel dashboard.

**Reply by quyenvnk63** [reply]
I already fixed it. the cause is that i have configured {
     "env": {
       "AWS_ACCESS_KEY_ID": "@aws-access-key-id",
       "AWS_SECRET_ACCESS_KEY": "@aws-secret-access-key",
       "AWS_DEFAULT_REGION": "@aws-default-region",
       "AWS_BUCKET": "@aws-bucket",
       "AWS_USE_PATH_STYLE_ENDPOINT": "@aws-use-path-style-endpoint"
     }
   } in the file vercle.json. I deleted it and it worked, although I don't understand the reason either.

**Reply by icyJoseph** [reply]
Probably out of sync with the dashboard value - don't forget to mark as resolved and close the thread :)
